packages/terra-form-field/docs/MigrationGuide.md

```diff
@@ -0,0 +1,76 @@
+# Migrating from terra-form's Field Component to terra-form-field
+
+The following sections describe how to migrate from the Field component provided in the terra-form package to the Field component provided in the terra-form-field package. Terra-form-field provides enhanced API and layout to begin the first phase of the OSC uplift.
+
+## Getting Started
+
+- Install with [npmjs](https://www.npmjs.com):
+  - `npm install terra-form-field`
+
+## Prop Changes
+#### Props Added
+- `errorIcon` - icon to indicates the Field is invalid.
+- `hideRequired` - indicates a Field is required, but hides the required indicator.
+- `isInvalid` - indicates whether or not a field is valid or invalid.
+- `isLabelHidden` - indicates whether or not the label is visible.
+- `showOptional` - indicates a Field is optional.
+
+#### Props Changed
+- `label` - optional --> required.
+
+## Behavior Changes
+#### Added/Changed Behavior
+- `isInvalid` - when true, the provided error icon and error text will show. Previously, the error text would alway show when provided.
+- `hideRequired` - when used, the required indicator is hidden, but when `isInvalid`, the required indicated will show.
+- `showOptional` - when used, the label '(optional)' is appended to the provide label text.
+- `isLabelHidden` - when used, the label is created on the DOM for accessibility, but the label is not visible.
+
+## Code Migration
+#### Migrating from terra-form/lib/Field to terra-form-field:
+```diff
+  import React from 'react';
+- import Field from 'terra-form/lib/Field';
++ import Field from 'terra-form-field';
+
+  <Field
+    label="Field Label"
+    labelAttrs={{ className: 'field-label' }}
+    error="This field is required"
+    help="This is a required field"
++   isInvalid={this.state.isInvalid}  // Needed to display error icon and error text
+    isInline
+    required
```

Comment:
this example here, I would make it take booleans here to show it matching the api

isInvalid={true}
isInline={false}
required={true}

Reply:
The default value of each of these boolean props are false, passing the name evaluates it to true. The convention with naming boolean props is to provide this shorthand prop passing. The example provided is how we'd expect one to use these props, unless they are toggling state like `isInvalid`.